tests/default/_core/search/knn_disk_based.yaml

```diff
@@ -0,0 +1,71 @@
+$schema: ../../../../json_schemas/test_story.schema.yaml
+
+description: Test search endpoint with knn query.
+version: '>= 2.17'
+
+prologues:
+  - method: PUT
+    path: /my-vector-index
+    request:
+      payload:
+        settings:
+          index:
+            knn: true
+        mappings:
+          properties:
+            my_vector_field:
+              type: knn_vector
+              dimension: 8
+              space_type: innerproduct
+              data_type: float
+              mode: on_disk
+              compression_level: 16x
+    status: [200]
+  - method: POST
+    path: /_bulk
+    request:
+      content_type: application/x-ndjson
+      payload:
+        -  { "index": { "_index": "my-vector-index", "_id": "1" } }
+        -  { "my_vector_field": [ 1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5 ], "price": 12.2 }
+        -  { "index": { "_index": "my-vector-index", "_id": "2" } }
+        -  { "my_vector_field": [ 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5 ], "price": 7.1 }
+        -  { "index": { "_index": "my-vector-index", "_id": "3" } }
+        -  { "my_vector_field": [ 3.5, 3.5, 3.5, 3.5, 3.5, 3.5, 3.5, 3.5 ], "price": 12.9 }
+        -  { "index": { "_index": "my-vector-index", "_id": "4" } }
+        -  { "my_vector_field": [ 4.5, 4.5, 4.5, 4.5, 4.5, 4.5, 4.5, 4.5 ], "price": 1.2 }
+        -  { "index": { "_index": "my-vector-index", "_id": "5" } }
+        -  { "my_vector_field": [ 5.5, 5.5, 5.5, 5.5, 5.5, 5.5, 5.5, 5.5 ], "price": 3.7 }
+        -  { "index": { "_index": "my-vector-index", "_id": "6" } }
+        -  { "my_vector_field": [ 6.5, 6.5, 6.5, 6.5, 6.5, 6.5, 6.5, 6.5 ], "price": 10.3 }
+        -  { "index": { "_index": "my-vector-index", "_id": "7" } }
+        -  { "my_vector_field": [ 7.5, 7.5, 7.5, 7.5, 7.5, 7.5, 7.5, 7.5 ], "price": 5.5 }
+        -  { "index": { "_index": "my-vector-index", "_id": "8" } }
+        -  { "my_vector_field": [ 8.5, 8.5, 8.5, 8.5, 8.5, 8.5, 8.5, 8.5 ], "price": 4.4 }
+        -  { "index": { "_index": "my-vector-index", "_id": "9" } }
+        -  { "my_vector_field": [ 9.5, 9.5, 9.5, 9.5, 9.5, 9.5, 9.5, 9.5 ], "price": 8.9 }
```

Comment:
We plan to use these examples in documentation, so maybe rename the fields to be movie/books based. For example instead of `my_vector_field` it could just be called `recommendation_vector`, and `price` could remain `price` or be `duration`.